build-aux/snap/snapcraft.yaml

```diff
@@ -28,172 +29,272 @@ license: GPL-3.0
+    source-type: git
+    <<:
+      - &dynamic-linker
+        build-environment:
+          - to amd64:
+              - DYNAMIC_LINKER: ld-linux-x86-64.so.2
+          - to s390x:
+              - DYNAMIC_LINKER: ld64.so.1
+          - to arm64:
+              - DYNAMIC_LINKER: ld-linux-aarch64.so.1
+          - to armhf:
+              - DYNAMIC_LINKER: ld-linux-armhf.so.3
+          - to riscv64:
+              - DYNAMIC_LINKER: ld-linux-riscv64-lp64d.so.1
+          - to ppc64el:
+              - DYNAMIC_LINKER: ld64.so.2
     override-build: |
-      # unset the LD_FLAGS and LD_LIBRARY_PATH vars that snapcraft sets for us
-      # as those will point to the $SNAPCRAFT_STAGE which on re-builds will 
-      # contain things like libc and friends that confuse the debian package
-      # build system
-      # TODO: should we unset $PATH to not include $SNAPCRAFT_STAGE too?
-      unset LD_FLAGS
-      unset LD_LIBRARY_PATH
-      # if we are root, disable tests because a number of them fail when run as
-      # root
-      if [ "$(id -u)" = "0" ]; then
-        DEB_BUILD_OPTIONS=nocheck
-        export DEB_BUILD_OPTIONS
-      fi
-      # run the real build (but just build the binary package, and don't
-      # bother compressing it too much)
-      dpkg-buildpackage -b -Zgzip -zfast -uc -us
-      dpkg-deb -x $(pwd)/../snapd_*.deb $SNAPCRAFT_PART_INSTALL
-      # not included in the deb as it's only used with UC20 preseeding.
-      cp -a data/preseed.json $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/
-      # Note that this check should run *after* dpkg-buildpackage was run
-      # as this will re-run "go generate" which may cause a dirty tree
-      #
-      # TODO: when something like "craftctl get-version" is ready, then we can
-      # use that, but until then, we have to re-run mkversion.sh to check if the
-      # version number was set as "dirty" from the override-pull step or during
-      # the build step
-      if sh -x ./mkversion.sh --output-only | grep "dirty"; then
-        mkdir -p $SNAPCRAFT_PART_INSTALL/usr/lib/snapd
-        (
-          echo "dirty git tree during build detected:"
-          git status
-          git diff
-        ) > $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/dirty-git-tree-info.txt
-      fi
-
-  # xdelta is used to enable delta downloads (even if the host does not have it)
-  xdelta3:
-    plugin: nil
-    stage-packages:
-      - xdelta3
+      rm -rf build
+      mkdir -p build
+      cd build
+      unset LDFLAGS
+      unset CPPFLAGS
+      export CFLAGS="-pipe -O2 -g"
+      cat <<EOF >configparms
+      libdir=/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}
+      slibdir=/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}
+      rtldir=/snap/snapd/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}
+      EOF
+      "${CRAFT_PART_SRC_WORK}/configure" --prefix=/snap/snapd/current/usr
+      make -j"${CRAFT_PARALLEL_BUILD_COUNT}" -C .. objdir="${PWD}" "${PWD}/linkobj/libc_pic.a"
```

Comment:
There's a bunch of patches in https://git.launchpad.net/ubuntu/+source/glibc/tree/debian/patches?h=ubuntu/jammy-updates -- are we applying them? I think we want to understand if we should/should not do so (my bet is yes, as that's how security fixes will land). Apologies if we do, this is unclear in the code.

Reply:
Ah, interesting. This repo is missing the `applied/` branches. And I am checking out the unapplied one. I guess I have to run quilt.

Reply:
Now `quilt push -a` is called.

Reply:
Yes, please apply all patches.